docs/DeveloperGuide.md

```diff
@@ -897,60 +898,41 @@ Priorities: High (must have) - `***`, Medium (nice to have) - `**`, Low (unlikel
-* When a new customer or delivery is created, the ID of the customer or delivery would be the first available ID from 1
-  to the maximum integer in the list of customers or deliveries respectively.
-* ID of a new customer and delivery will reset to 1 for customers and deliveries if the user deletes all
-  customers and deliveries using the `clear` command.
-* ID of a new customer and delivery will reset to 1 for customers and deliveries if the user deletes his account.
+2. Currently, the `find` command for customer would allow special characters to be used to search for the name of a
+   customer. We plan to disallow special characters as a user cannot add special characters to the name of a customer in
+   the first place.
 
-#### Find Customer Command
+3. Currently, you would need to type `delivery list` or `customer list` in order to switch between the two lists which
+   may hinder the user's ability to quickly type commands. We can create two side-by-side list views of both customer
+   and delivery for quicker reference and enable users to make quicker commands.
 
-* Find command for customer would disallow special characters as a user cannot add special characters to the name of a
-  customer.
+4. Currently, some error messages are not specific enough. Error messages can be made more specific and instructive. For
+   example, if a user enters 2023-02-30, the error message
+   can be "Invalid date. The date entered does not exist.", however the current implementation only states "Dates
+   should be in the format yyyy-MM-dd". Or, if a user enters the incorrect ID, the error should show "Customer ID" or "
+   Delivery ID", however, the current implementation does not specify whether it is for the customer ID or for delivery
+   ID and only states "ID". Or, if a user inputs a negative value or zero for a numerical parameter that requires a
+   positive integer, the error message should be more specific and state "Please enter a positive integer." instead of "
+   Invalid Command Format".
 
-#### Delivery Status
+5. Currently, only the user password is the only thing that is hashed in the authentication file. We plan to hash the
+   whole JSON authentication file and the data to prevent unauthorised access to the user's account and data
+   instead of only hashing the user password.
 
-* Allow the user to set their own customised delivery status.
+6. Currently, you can recover your account while logged in. We plan to disallow the user to recover their account while
+   logged in.
 
-#### Enhanced Delivery Address
+7. Currently, you can put duplicate prefixes for delivery list to filter or sort the deliveries. We plan to disallow
+   duplicate prefixes for delivery list.
 
-* Allow tagging by area and region name to delivery address and enable users to filter them accordingly.
+8. Currently, you can filter delivery list by a customer id that does not exist. We plan to disallow filtering delivery
+   list by a customer id that does not exist.
```

Comment:
Should we mention being able to use certain special characters for certain fields, e.g. in name s/o or in notes punctuation like "." and ","